File: app/src/main/java/io/trewartha/positional/CoordinatesFragment.java
package io.trewartha.positional;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

public abstract class CoordinatesFragment extends Fragment {

    protected final String STATE_LATITUDE = "latitude";
    protected final String STATE_LONGITUDE = "longitude";

    protected double latitude;
    protected double longitude;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (savedInstanceState != null) {
            latitude = savedInstanceState.getDouble(STATE_LATITUDE);
            longitude = savedInstanceState.getDouble(STATE_LONGITUDE);
        }
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putDouble(STATE_LATITUDE, latitude);
        outState.putDouble(STATE_LONGITUDE, longitude);
    }

    public abstract void setCoordinates(double latitude, double longitude);
}

File: app/src/main/java/io/trewartha/positional/position/CoordinatesFragment.kt
package io.trewartha.positional.position

import android.os.Bundle
import android.support.v4.app.Fragment

abstract class CoordinatesFragment : Fragment() {

    companion object {
        private const val STATE_LATITUDE = "latitude"
        private const val STATE_LONGITUDE = "longitude"
    }

    protected var latitude: Double = 0.0
    protected var longitude: Double = 0.0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        latitude = savedInstanceState?.getDouble(STATE_LATITUDE) ?: 0.0
        longitude = savedInstanceState?.getDouble(STATE_LONGITUDE) ?: 0.0
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putDouble(STATE_LATITUDE, latitude)
        outState.putDouble(STATE_LONGITUDE, longitude)
    }

    abstract fun setCoordinates(latitude: Double, longitude: Double)
}
